pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Markers.java

```diff
@@ -290,6 +306,15 @@ private static ByteBuf newTxnMarker(MarkerType markerType, long sequenceId, long
         MessageMetadata msgMetadata = msgMetadataBuilder.build();
 
         ByteBuf payload = PooledByteBufAllocator.DEFAULT.buffer();
+        if (messageIdData.isPresent()) {
+            PulsarMarkers.TxnCommitMarker commitMarker =
+                PulsarMarkers.TxnCommitMarker.newBuilder().setMessageId(messageIdData.get()).build();
+            int size = commitMarker.getSerializedSize();
+            payload.release();
```

Comment:
I would recommend avoiding unnecessary allocations and release.

```
ByteBuf payload;
if (messageIdData.isPresent()) {
    PulsarMarkers.TxnCommitMarker commitMarker = ...;
    ...
    payload = PooledByteBufAllocator.DEFAULT.buffer(size);
    ...
} else {
    payload = PooledByteBufAllocator.DEFAULT.buffer();
}
```

Reply:
fixed